### ISSUE TITLE
ICONV/OCONV matches shortest first, unlike 1.4.0

### ISSUE THREAD

**Issue by changwoo** [hunspell/hunspell#480]
In 1.4.0, ICONV/OCONV feature was finding the longest matched pattern. But the 1.6.0 version seems to find the shortest matched one. This OCONV behavior change breaks Korean dictionary.

Short example:

```
$ cat oconv-length.aff
ICONV 2
ICONV Ga GA
ICONV Gag GAG
$ cat oconv-length.dic
cat oconv-length.dic
1
GAG
$
```
In hunspell 1.4.0,

```
$ hunspell --version|grep Hunspell 
@(#) International Ispell Version 3.2.06 (but really Hunspell 1.4.0)
$ echo Gag | hunspell -d oconv-length
Hunspell 1.4.0
*
```

But in the 1.6.0 version,

```
$ echo Gag | hunspell -d oconv-length
Hunspell 1.6.0
& Gag 1 0: GAG

$
```

**Comment by changwoo** [comment]
It was version 1.6.0 change.

**Comment by dimztimz** [comment]
Probably related to this commit 73b1cad1af7ab94252f75784fa6724cf062a6966, which was part of PR #376. Can you test the behavior before and after the commit?

**Comment by changwoo** [comment]
Yes, before the commit it worked OK,

```
$ echo Gag | ./hunspell -d oconv-length
Hunspell 1.4.0
*
```

After the commit, actually hunspell crashed,  After the fix in hunspell/hunspell#376, the match behavior changed like the current one.

```
$ echo Gag | ./hunspell -d oconv-length
Hunspell 1.5.0
& Gag 1 0: GAG
```
